### PR TITLE
multi pub/sub support

### PR DESCRIPTION
**Pull request by mchmarny**
Implements multi pub/sub support outlined in #31

### CODE REVIEW

File: service/grpc/topic.go [hunk 1]
```diff
@@ -40,10 +50,11 @@ func (s *Server) AddTopicEventHandlerWithMetadata(topic string, m map[string]str
 // To subscribe to a topic named TopicA
```
Comment (yaron2) [inline service/grpc/topic.go:50]: Is this the right comment for what this method does?

Comment (yaron2) [reply service/grpc/topic.go:50]: Resolved

File: client/pubsub.go [hunk 2]
```diff
@@ -8,17 +8,21 @@ import (
 )
 
 // PublishEvent pubishes data onto specific pubsub topic.
-func (c *GRPCClient) PublishEvent(ctx context.Context, topic string, in []byte) error {
+func (c *GRPCClient) PublishEvent(ctx context.Context, component, topic string, in []byte) error {
 	if topic == "" {
```
Comment (yaron2) [inline client/pubsub.go:12]: Might want to make this the same as `AddTopicEventHandler` and be consistent with the validation messages:

```
if topic == "" {
		return fmt.Errorf("topic name required")
	}
	if component == "" {
		return fmt.Errorf("component name required")
	}
```

Comment (mchmarny) [reply client/pubsub.go:12]: Validation messages are now consistent

File: client/pubsub.go [hunk 3]
```diff
@@ -10,13 +10,13 @@ import (
...
-		return errors.New("nil topic")
+		return errors.New("topic name required")
 	}
 	if component == "" {
-		return errors.New("nil component")
+		return errors.New("component name required")
 	}
 	if in == nil {
-		return errors.New("nil in")
+		return errors.New("in data required")
```
Comment (yaron2) [inline client/pubsub.go:19]: Data is not a requirement, empty topics are valid.

**Pull request merged**